src/interpolation/details/ArborX_InterpDetailsMovingLeastSquaresCoefficients.hpp

```diff
@@ -26,13 +26,206 @@
 namespace ArborX::Interpolation::Details
 {
 
+template <typename CRBFunc, typename PolynomialDegree, typename SourcePoints,
+          typename TargetAccess, typename Coefficients, typename ExecutionSpace>
+class MovingLeastSquaresCoefficientsKernel
+{
+private:
+  using ScratchMemorySpace = typename ExecutionSpace::scratch_memory_space;
+
+  using SourcePoint = typename SourcePoints::non_const_value_type;
+  using TargetPoint = typename TargetAccess::value_type;
+
+  using CoefficientsType = typename Coefficients::non_const_value_type;
+
+  static constexpr int dimension = GeometryTraits::dimension_v<SourcePoint>;
+  static constexpr int degree = PolynomialDegree::value;
+  static constexpr int poly_size = polynomialBasisSize<dimension, degree>();
+
+  template <typename T>
+  using UnmanagedView =
+      Kokkos::View<T, ScratchMemorySpace, Kokkos::MemoryUnmanaged>;
+
+  using LocalSourcePoints = Kokkos::Subview<SourcePoints, int, Kokkos::ALL_t>;
+  using LocalPhi = UnmanagedView<CoefficientsType *>;
+  using LocalVandermonde = UnmanagedView<CoefficientsType **>;
+  using LocalMoment = UnmanagedView<CoefficientsType **>;
+  using LocalSVDDiag = UnmanagedView<CoefficientsType *>;
+  using LocalSVDUnit = UnmanagedView<CoefficientsType **>;
+  using LocalCoefficients = Kokkos::Subview<Coefficients, int, Kokkos::ALL_t>;
```

Comment:
Why is it relevant to say "local" here?

Reply:
It is to signify "local to the current target"